NeuroID/NeuroIDClass/Extensions/NIDUser.swift

```diff
@@ -94,61 +88,59 @@ public extension NeuroID {
             saveEventToLocalDataStore(sessionIdEvent)
         }
     }
-
-    static func setUserID(_ userId: String) -> Bool {
-        // Redundant check to ensure CURRENT_ORIGIN is never unsafely accessed
-        if CURRENT_ORIGIN == nil {
-            CURRENT_ORIGIN = SessionOrigin.NID_ORIGIN_CUSTOMER_SET.rawValue
-            CURRENT_ORIGIN_CODE = SessionOrigin.NID_ORIGIN_CODE_CUSTOMER.rawValue
+    
+    internal static func getOriginResult(idValue: String,
+                                         validID: Bool,
+                                         userGenerated: Bool,
+                                         idType: UserIDTypes) -> SessionIDOriginalResult{
+        let origin = userGenerated ? SessionOrigin.NID_ORIGIN_CUSTOMER_SET.rawValue : SessionOrigin.NID_ORIGIN_NID_SET.rawValue
+        var originCode = SessionOrigin.NID_ORIGIN_CODE_FAIL.rawValue
+        if validID {
+            originCode =  userGenerated ? SessionOrigin.NID_ORIGIN_CODE_CUSTOMER.rawValue : SessionOrigin.NID_ORIGIN_CODE_NID.rawValue
         }
-        let res = setGenericUserID(
-            userId: userId, type: .userID
-        ) { success in
-            if success {
-                sendOriginEvent(origin: CURRENT_ORIGIN!, originCode: CURRENT_ORIGIN_CODE!, originSessionID: userId)
-                NeuroID.userID = userId
-            }
-
-            return success
+        return SessionIDOriginalResult(origin: origin,originCode: originCode,idValue: idValue,idType: idType)
+    }
+    
+    static func setUserID(_ userId: String, _ userGenerated: Bool? = true)-> Bool{
```

Comment:
Can we have an internal method that is this signature and a public version that is the id string only? Customers should not be able to set a `userGenerated` boolean even by accident. That will also allow the `userGenerated` to be non-optional